File: leetcode/0590. N-ary Tree Postorder Traversal/Solution.swift
// https://leetcode.com/problems/n-ary-tree-postorder-traversal/

public class TreeNode {
    public var val: Int
    public var children: [TreeNode?]
    public init(_ val: Int) {
        self.val = val
        self.children = []
    }
}

class Solution {
    func postorder(_ root: TreeNode?) -> [Int] {
        guard let root = root else { return [] }

        var r = [Int]()
        for child in root.children {
            r += postorder(child)
        }

        return r + [root.val]
    }
}

File: leetcode/0590. N-ary Tree Postorder Traversal/Solution.py
# https://leetcode.com/problems/n-ary-tree-postorder-traversal/

"""
# Definition for a Node.
class Node:
    def __init__(self, val, children):
        self.val = val
        self.children = children
"""

class Solution:
    def postorder(self, root: 'Node') -> List[int]:
        if root is None:
            return []
        
        r = []
        for child in root.children:
            r += self.postorder(child)
        
        return r + [root.val]